cmd/otelcorecol/main_windows.go

```diff
@@ -7,42 +7,25 @@ package main
 
 import (
 	"fmt"
-	"os"
+	"syscall"
 
+	"golang.org/x/sys/windows"
 	"golang.org/x/sys/windows/svc"
 
 	"go.opentelemetry.io/collector/otelcol"
 )
 
 func run(params otelcol.CollectorSettings) error {
-	if useInteractiveMode, err := checkUseInteractiveMode(); err != nil {
-		return err
-	} else if useInteractiveMode {
-		return runInteractive(params)
-	} else {
-		return runService(params)
-	}
-}
-
-func checkUseInteractiveMode() (bool, error) {
-	// If environment variable NO_WINDOWS_SERVICE is set with any value other
-	// than 0, use interactive mode instead of running as a service. This should
-	// be set in case running as a service is not possible or desired even
-	// though the current session is not detected to be interactive
-	if value, present := os.LookupEnv("NO_WINDOWS_SERVICE"); present && value != "0" {
-		return true, nil
-	}
-
-	isInteractiveSession, err := svc.IsAnInteractiveSession()
-	if err != nil {
-		return false, fmt.Errorf("failed to determine if we are running in an interactive session: %w", err)
-	}
-	return isInteractiveSession, nil
-}
-
-func runService(params otelcol.CollectorSettings) error {
-	// do not need to supply service name when startup is invoked through Service Control Manager directly
+	// No need to supply service name when startup is invoked through
+	// the Service Control Manager directly.
 	if err := svc.Run("", otelcol.NewSvcHandler(params)); err != nil {
+		errno, ok := err.(syscall.Errno)
```

Comment:
I think you should use this as `if errors.Is(err, windows. ERROR_FAILED_SERVICE_CONTROLLER_CONNECT)`

Reply:
d77a32989